### PR TITLE
Option to skip through the video by scrolling

### PR DESCRIPTION
**Pull request by vr10t**
---
Option to skip through the video by scrolling
---

**Important note**
We may remove your pull request if you do not use this provided PR template correctly.

**Pull Request Type**
Please select what type of pull request this is:
- [ ] Bugfix
- [x] Feature Implementation

**Related issue**
Please link the issue your pull request is referring to. If this pull request fully resolves the relevant issue, put "closes" before the issue number. Example: "closes #123456".

**Description**
Add setting to skip through the video by scrolling on the player, just like MPV, which is my favourite player. You can skip 5s ahead by scrolling up and 5s behind by scrolling down.
**Screenshots (if appropriate)**
Please add before and after screenshots if there is a visible change.

**Testing (for code that is not small enough to be easily understandable)**
Has this pull request been tested?
Please describe shortly how you tested it and whether there are any ramifications remaining. 

**Desktop (please complete the following information):**
 - OS: [Manjaro]
 - OS Version: [5.10.117-1]
 - FreeTube version: [0.17.0]

**Additional context**

### CODE REVIEW

File: src/renderer/components/ft-video-player/ft-video-player.js [hunk 1]
```diff
@@ -693,6 +700,19 @@ export default Vue.extend({
       }
     },
 
+    mouseScrollSkip: function (event) {
+      if (event.target && (event.target.matches('.vjs-tech') || event.target.matches('.ftVideoPlayer'))) {
```
Comment (PikachuEXE) [inline src/renderer/components/ft-video-player/ft-video-player.js:704]: Maybe add some code comment to explain the selectors?

**Review by PikachuEXE — COMMENTED** [review]
Can you explain more about "scrolling on the player" since I see two selectors

File: src/renderer/components/ft-video-player/ft-video-player.js (hunk 1 above)
Comment (vr10t) [reply src/renderer/components/ft-video-player/ft-video-player.js:704]: To be honest I have no idea why, but targeting just 'vjs-tech' doesn't seem to work all the time. 99% of the time it works just fine  but in some rare cases '.ftVideoPlayer'  gets targeted on hover and I don't know why. I added it as a selector just to account for that rare case. If I can figure out what's causing it maybe I can remove it.

**Comment by PikachuEXE** [comment]
Anyone wanna take over this? @ChunkyProgrammer ?

**Comment by ChunkyProgrammer** [comment]
Made it so you can only enable one of the `scroll over player` settings

**Comment by PikachuEXE** [comment]
Things to be tested:
- Play video with new setting **disabled**, scrolling shouldn't do anything
- Play video with new setting **enabled**, scrolling should skip video by N seconds (default 5, settings somewhere)
- Setting `Skip by Scrolling Over Video Player` and `Scroll Volume Over Video Player` cannot be enabled at the same time (enable one the other one will be disabled)

th

**Review by PikachuEXE — APPROVED** [review]
Forgot to mark as reviewed

File: src/renderer/components/ft-video-player/ft-video-player.js [hunk 2]
```diff
@@ -693,6 +700,19 @@ export default Vue.extend({
...
+        event.preventDefault()
+
+        if (event.wheelDelta > 0) {
+          this.changeDurationBySeconds(this.defaultSkipInterval * this.player.playbackRate())
+        }
+        if (event.wheelDelta < 0) {
+          this.changeDurationBySeconds(-this.defaultSkipInterval * this.player.playbackRate())
+        }
+      }
```
[the remark this answers is not in the document]
Comment (PikachuEXE) [reply src/renderer/components/ft-video-player/ft-video-player.js:742]: Oh my old pending comment...
Can be ignored

**Comment by PikachuEXE** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc @absidue @ChunkyProgrammer ?_?

**Review by ChunkyProgrammer — APPROVED** [review]

**Review by absidue — APPROVED** [review]

**Comment by absidue** [comment]
I only checked the code and forgot to test it before approving.

When you have "Scroll playback rate over video player" and "Skip by Scrolling Over Video Player" and scroll while holding ctrl it should only change the playback speed, instead it changes the playback speed and seeks.

**Comment by PikachuEXE** [comment]
@absidue fixed in https://github.com/FreeTubeApp/FreeTube/pull/2989

**Pull request merged**